---
title: C26115
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C26115
helpviewer_keywords:
- C26115
ms.assetid: 3977a2bb-d1fe-4510-89dd-07fdc69e911c
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: 9523129811d94491062b8c337d81dd8143e50dcf
ms.sourcegitcommit: 535ef05b1e553f0fc66082cd2e0998817eb2a56a
ms.translationtype: MT
ms.contentlocale: zh-CN
ms.lasthandoff: 10/07/2019
ms.locfileid: "72015681"
---
# <a name="c26115"></a>C26115
警告 C26115：未能释放 @no__t > 函数 \<func > 中的锁。

 C/C++ program 中的语义范围内锁*获取*和锁*释放*对不是由语言执行的。 通过对并发状态进行显著的修改，函数可能会造成锁定副作用。 例如，锁包装器函数会增加给定锁的锁获取数或锁计数。

 通过分别使用 `_Acquires_lock_` 或 `_Releases_lock_`，可批注由于锁获取或锁释放而产生副作用的函数。 如果没有此类批注，函数在返回之后，预计不会更改任何锁计数。 如果获取和释放不平衡，它们将被视为*孤立*。 当函数引入孤立锁时，将发出警告 C26115。

## <a name="example"></a>示例
 下面的示例将生成警告 C26115，因为未使用 `_Acquires_lock_` 进行批注的函数中存在孤立锁。

```

typedef struct _DATA
{
    CRITICAL_SECTION cs;
} DATA;

void FailToReleaseLock(int flag, DATA* p)
{
    EnterCriticalSection(&p->cs);

    if (flag)
        return; // Warning C26115

    LeaveCriticalSection(&p->cs);
}
```